pkg/resizer/csi_resizer.go

```diff
@@ -120,14 +130,32 @@ func (r *csiResizer) CanSupport(pv *v1.PersistentVolume) bool {
 	return true
 }
 
+// Resize resizes the persistence volume given request size
+// It supports both CSI volume and migrated in-tree volume
 func (r *csiResizer) Resize(pv *v1.PersistentVolume, requestSize resource.Quantity) (resource.Quantity, bool, error) {
 	oldSize := pv.Spec.Capacity[v1.ResourceStorage]
 
-	source := pv.Spec.CSI
-	if source == nil {
-		return oldSize, false, errors.New("not a CSI volume")
+	var volumeID string
+	var source *v1.CSIPersistentVolumeSource
+	if pv.Spec.CSI != nil {
+		// handle CSI volume
+		source = pv.Spec.CSI
+		volumeID = source.VolumeHandle
+	} else {
+		if csitranslationlib.IsMigratedCSIDriverByName(r.name) {
+			// handle migrated in-tree volume
+			csiPV, err := csitranslationlib.TranslateInTreePVToCSI(pv)
+			if err != nil {
+				return oldSize, false, fmt.Errorf("failed to translate persistent volume: %v", err)
+			}
+			source = csiPV.Spec.CSI
+			volumeID = source.VolumeHandle
+		} else {
+			// non-migrated in-tree volume
+			return oldSize, false, fmt.Errorf("volume %v is not migrated to CSI", pv.Name)
```

Comment:
How do we get here? The annotation is set with the name that the csi plugin is configured for but the migration library doesn't support it?

Reply:
This will be an in-tree PV that doesn't support CSI migration yet. This should be already gated by `CanSupport` check earlier, but still adding the else branch for the sake of in case this happens unintentionally, or because of some bug